### ISSUE TITLE
bird: Netlink: Network is down

### ISSUE THREAD

**Issue by dotbalo** [projectcalico/calico#3134]
I have a k8s cluster with version 1.13.6. Then I added a node and the cluster seemed to be working fine, but calico had an error: bird: Netlink: Network is down. Only this new node has this error, the other nodes are normal, then this new node cannot connect to the pods of other nodes, nor can it resolve svc.My calico was upgraded from 3.6.1 to 3.6.5.

## Expected Behavior
I want to solve the problem of bird: Netlink: Network is down

## Current Behavior
Cluster status:
````
[root@k8s-master01 v3.6.5]# kubectl get node
NAME                STATUS   ROLES   AGE    VERSION
k8s-master01.xiqu   Ready    node    243d   v1.13.6
k8s-master02.xiqu   Ready    node    147m   v1.13.6
k8s-master03.xiqu   Ready    node    243d   v1.13.6
k8s-node01.xiqu     Ready    node    243d   v1.13.6
k8s-node02.xiqu     Ready    node    243d   v1.13.6

k8s-master02.xiqu is new node.
````

New Node Route info:
````
[root@k8s-master02 ~]# ip route
default via 10.103.236.254 dev enp7s0f0 proto static metric 100 
10.103.236.0/24 dev enp7s0f0 proto kernel scope link src 10.103.236.178 metric 100 
169.254.0.0/16 dev enp7s0f0 scope link metric 1002 
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown 
blackhole 177.245.72.64/26 proto bird 
177.245.72.68 dev cali34878b38246 scope link 
177.245.72.70 dev cali74e0d914f9c scope link
````

Then I started a container on the new node, and then this container could not resolve the Kubernetes service, and could not communicate with the Pods of other nodes.
````
[root@k8s-master01 v3.6.5]# kubectl get po -owide
NAME                            READY   STATUS    RESTARTS   AGE   IP                NODE                NOMINATED NODE   READINESS GATES
fortio-deploy-8ddb4b96d-cckv5   1/1     Running   0          46d   177.253.190.121   k8s-master03.xiqu   <none>           <none>
redis-trib-7f84cc4c56-t8q7f     1/1     Running   0          52m   177.245.72.70     k8s-master02.xiqu   <none>           <none>
[root@k8s-master01 v3.6.5]# kubectl exec -ti redis-trib-7f84cc4c56-t8q7f  -- nslookup kubernetes
nslookup: can't resolve '(null)': Name does not resolve

nslookup: can't resolve 'kubernetes': Try again
command terminated with exit code 1

````

## Context
````
My newly added node cannot communicate with the pods of other nodes, hoping to solve this problem.
````

## Your Environment
* Calico version  --> 3.6.5
* Orchestrator version (e.g. kubernetes, mesos, rkt):
````
[root@k8s-master01 v3.6.5]# kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.6", GitCommit:"abdda3f9fefa29172298a2e42f5102e777a8ec25", GitTreeState:"clean", BuildDate:"2019-05-08T13:53:53Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.6", GitCommit:"abdda3f9fefa29172298a2e42f5102e777a8ec25", GitTreeState:"clean", BuildDate:"2019-05-08T13:46:28Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
````
* Operating System and version: 
````
[root@k8s-master02 ~]# cat /etc/redhat-release 
CentOS Linux release 7.7.1908 (Core)
You have new mail in /var/spool/mail/root
[root@k8s-master02 ~]# uname 
Linux
[root@k8s-master02 ~]# uname -a
Linux k8s-master02.xiqu 4.18.9-1.el7.elrepo.x86_64 #1 SMP Thu Sep 20 09:04:54 EDT 2018 x86_64 x86_64 x86_64 GNU/Linux
````

* calico status
````
[root@k8s-master01 v3.6.5]# kubectl exec -ti calicoctl-6c6fbb8749-snxs7  -n !$ -- calicoctl get node -owide
kubectl exec -ti calicoctl-6c6fbb8749-snxs7  -n kube-system -- calicoctl get node -owide
NAME                ASN         IPV4                IPV6   
k8s-master01.xiqu   (unknown)   10.103.236.177/24          
k8s-master02.xiqu   (unknown)   10.103.236.178/24          
k8s-master03.xiqu   (unknown)   10.103.236.179/24          
k8s-node01.xiqu     (unknown)   10.103.236.175/24          
k8s-node02.xiqu     (unknown)   10.103.236.176/24
````

Calico logs:
````
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
2020-01-15 10:49:49.077 [INFO][76] health.go 150: Overall health summary=&health.HealthReport{Live:true, Ready:true}
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
2020-01-15 10:49:53.933 [INFO][76] int_dataplane.go 751: Applying dataplane updates
2020-01-15 10:49:53.933 [INFO][76] ipsets.go 223: Asked to resync with the dataplane on next update. family="inet"
2020-01-15 10:49:53.933 [INFO][76] ipsets.go 254: Resyncing ipsets with dataplane. family="inet"
2020-01-15 10:49:53.944 [INFO][76] ipsets.go 304: Finished resync family="inet" numInconsistenciesFound=0 resyncDuration=11.101272ms
2020-01-15 10:49:53.945 [INFO][76] int_dataplane.go 765: Finished applying updates to dataplane. msecToApply=11.640496
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
2020-01-15 10:49:55.165 [INFO][76] health.go 150: Overall health summary=&health.HealthReport{Live:true, Ready:true}
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
bird: Netlink: Network is down
bird: Netlink: Network is down
````

**Comment by tmjd** [comment]
Is NetworkManager running on the new host?

Have you tried monitoring the interfaces on the new host, is the interface going up and down?

Can you try pod to pod communication between pods on one of the previous hosts to this new node? Please make sure you use the pod IP to avoid involving DNS in the test.

**Comment by dotbalo** [comment]
@tmjd Thanks for your answer.

NetworkManager is down on the new node
````
[root@k8s-master02 ~]# systemctl status NetworkManager
● NetworkManager.service - Network Manager
   Loaded: loaded (/usr/lib/systemd/system/NetworkManager.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
     Docs: man:NetworkManager(8)
````

The newly added node seems to be normal, the status is Ready, and the Pod can be successfully created, but the network is not accessible. I pinged the Pod IP of the new node from the previous node and it did not work the other way around. 

I checked the routing table of the new node and did not generate routing information, so it was not working.
````
[root@k8s-master02 ~]# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.103.236.254  0.0.0.0         UG    0      0        0 enp7s0f0
10.103.236.0    0.0.0.0         255.255.255.0   U     0      0        0 enp7s0f0
169.254.0.0     0.0.0.0         255.255.0.0     U     1002   0        0 enp7s0f0
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
177.245.72.64   0.0.0.0         255.255.255.192 U     0      0        0 *
177.245.72.65   0.0.0.0         255.255.255.255 UH    0      0        0 cali12d4a061371
````

Newly added nodes have not encountered similar problems before. This node was originally a node in the cluster, and then a system failure occurred. Then I deleted the node from the cluster. After I reinstalled the system and rejoined the cluster, this problem occurred.

**Comment by tmjd** [comment]
> I pinged the Pod IP of the new node from the previous node and it did not work the other way around.
Could you expand on this? You said Pod IP of the new node what do you mean by that?
What traffic is working and not working between your nodes? If there is a problem where your nodes are having trouble communicating then that will certainly result in a problem with the pods being able to communicate.

**Comment by hhstu** [comment]
``` json
[root@ks-master1 ~]#  systemctl status NetworkManager
● NetworkManager.service - Network Manager
   Loaded: loaded (/usr/lib/systemd/system/NetworkManager.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2020-09-23 18:05:46 CST; 3 days ago
     Docs: man:NetworkManager(8)
 Main PID: 6066 (NetworkManager)
    Tasks: 3
   Memory: 9.4M
   CGroup: /system.slice/NetworkManager.service
           └─6066 /usr/sbin/NetworkManager --no-daemon

Sep 27 14:48:52 ks-master1 NetworkManager[6066]: <info>  [1601189332.2487] manager: (calico_tmp_B): new Veth device (/org/freedesktop/NetworkManager/Devices/48)
Sep 27 14:48:52 ks-master1 NetworkManager[6066]: <info>  [1601189332.2522] manager: (calico_tmp_A): new Veth device (/org/freedesktop/NetworkManager/Devices/49)
Sep 27 14:54:13 ks-master1 NetworkManager[6066]: <info>  [1601189653.5505] manager: (calico_tmp_B): new Veth device (/org/freedesktop/NetworkManager/Devices/50)
Sep 27 14:54:13 ks-master1 NetworkManager[6066]: <info>  [1601189653.5545] device (calico_tmp_A): driver '(null)' does not support carrier detection.
Sep 27 14:54:13 ks-master1 NetworkManager[6066]: <info>  [1601189653.5547] device (calico_tmp_A): driver 'veth' does not support carrier detection.
Sep 27 14:54:13 ks-master1 NetworkManager[6066]: <info>  [1601189653.5581] manager: (calico_tmp_A): new Veth device (/org/freedesktop/NetworkManager/Devices/51)
Sep 27 15:07:31 ks-master1 NetworkManager[6066]: <info>  [1601190451.1519] manager: (calico_tmp_B): new Veth device (/org/freedesktop/NetworkManager/Devices/52)
Sep 27 15:07:31 ks-master1 NetworkManager[6066]: <info>  [1601190451.1549] manager: (calico_tmp_A): new Veth device (/org/freedesktop/NetworkManager/Devices/53)
Sep 27 15:11:49 ks-master1 NetworkManager[6066]: <info>  [1601190709.9523] manager: (calico_tmp_B): new Veth device (/org/freedesktop/NetworkManager/Devices/54)
Sep 27 15:11:49 ks-master1 NetworkManager[6066]: <info>  [1601190709.9559] manager: (calico_tmp_A): new Veth device (/org/freedesktop/NetworkManager/Devices/55)
```

**Comment by hhstu** [comment]
calico-node    logs：

``` json

2020-09-27 07:27:05.308 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=0.152712
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
2020-09-27 07:27:11.516 [INFO][57] int_dataplane.go 907: Applying dataplane updates
2020-09-27 07:27:11.517 [INFO][57] ipsets.go 223: Asked to resync with the dataplane on next update. family="inet"
2020-09-27 07:27:11.517 [INFO][57] ipsets.go 306: Resyncing ipsets with dataplane. family="inet"
2020-09-27 07:27:11.525 [INFO][57] ipsets.go 356: Finished resync family="inet" numInconsistenciesFound=0 resyncDuration=8.50626ms
2020-09-27 07:27:11.526 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=8.953452
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: ...
2020-09-27 07:27:15.279 [INFO][57] ipip_mgr.go 133: Tunnel wasn't admin up, enabling it flags=0 mtu=1440 tunnelAddr=10.233.77.0
2020-09-27 07:27:15.279 [INFO][57] ipip_mgr.go 138: Set tunnel admin up mtu=1440 tunnelAddr=10.233.77.0
2020-09-27 07:27:15.280 [INFO][57] int_dataplane.go 551: Linux interface state changed. ifaceName="tunl0" state="up"
2020-09-27 07:27:15.280 [INFO][57] ipip_mgr.go 183: Address wasn't present, adding it. addr=10.233.77.0 link="tunl0"
2020-09-27 07:27:15.280 [INFO][57] int_dataplane.go 797: Received interface update msg=&intdataplane.ifaceUpdate{Name:"tunl0", State:"up"}
2020-09-27 07:27:15.280 [INFO][57] int_dataplane.go 907: Applying dataplane updates
2020-09-27 07:27:15.281 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=0.10069800000000001
2020-09-27 07:27:15.281 [INFO][57] int_dataplane.go 584: Linux interface addrs changed. addrs=set.mapSet{"10.233.77.0":set.empty{}} ifaceName="tunl0"
2020-09-27 07:27:15.281 [INFO][57] iface_monitor.go 176: Netlink address update. addr="10.233.77.0" exists=true ifIndex=39
2020-09-27 07:27:15.281 [INFO][57] int_dataplane.go 812: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"tunl0", Addrs:set.mapSet{"10.233.77.0":set.empty{}}}
2020-09-27 07:27:15.281 [INFO][57] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"tunl0", Addrs:set.mapSet{"10.233.77.0":set.empty{}}}
2020-09-27 07:27:15.281 [INFO][57] ipsets.go 119: Queueing IP set for creation family="inet" setID="this-host" setType="hash:ip"
2020-09-27 07:27:15.281 [INFO][57] int_dataplane.go 907: Applying dataplane updates
2020-09-27 07:27:15.282 [INFO][57] ipsets.go 749: Doing full IP set rewrite family="inet" numMembersInPendingReplace=5 setID="this-host"
2020-09-27 07:27:15.283 [INFO][57] iface_monitor.go 176: Netlink address update. addr="10.233.77.0" exists=false ifIndex=39
2020-09-27 07:27:15.283 [INFO][57] int_dataplane.go 584: Linux interface addrs changed. addrs=set.mapSet{} ifaceName="tunl0"
2020-09-27 07:27:15.292 [INFO][57] int_dataplane.go 551: Linux interface state changed. ifaceName="tunl0" state="down"
2020-09-27 07:27:15.296 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=14.370695
2020-09-27 07:27:15.296 [INFO][57] int_dataplane.go 812: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"tunl0", Addrs:set.mapSet{}}
2020-09-27 07:27:15.296 [INFO][57] hostip_mgr.go 84: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"tunl0", Addrs:set.mapSet{}}
2020-09-27 07:27:15.296 [INFO][57] ipsets.go 119: Queueing IP set for creation family="inet" setID="this-host" setType="hash:ip"
2020-09-27 07:27:15.296 [INFO][57] int_dataplane.go 907: Applying dataplane updates
2020-09-27 07:27:15.298 [INFO][57] ipsets.go 749: Doing full IP set rewrite family="inet" numMembersInPendingReplace=4 setID="this-host"
2020-09-27 07:27:15.383 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=87.06132199999999
2020-09-27 07:27:15.384 [INFO][57] int_dataplane.go 797: Received interface update msg=&intdataplane.ifaceUpdate{Name:"tunl0", State:"down"}
2020-09-27 07:27:15.384 [INFO][57] int_dataplane.go 907: Applying dataplane updates
2020-09-27 07:27:15.384 [INFO][57] int_dataplane.go 921: Finished applying updates to dataplane. msecToApply=0.108008
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down
bird: Netlink: Network is down

```

**Comment by dotbalo** [comment]
There has been no solution to this problem. I tried many methods but couldn't solve the problem. In the end, I gave up.Everything looks normal, and bird can connect normally.
````
# netstat -lantup | grep bird
tcp        0      0 0.0.0.0:179             0.0.0.0:*               LISTEN      14035/bird          
tcp        0      0 192.168.1.230:48429    192.168.1.176:179      ESTABLISHED 14035/bird          
tcp        0      0 192.168.1.230:57359    192.168.1.178:179      ESTABLISHED 14035/bird          
tcp        0      0 192.168.1.230:40095    192.168.1.175:179      ESTABLISHED 14035/bird          
tcp        0      0 192.168.1.230:60001    192.168.1.179:179      ESTABLISHED 14035/bird
...
````

But there is no calico interface.
````
# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         192.168.1.254  0.0.0.0         UG    100    0        0 ens192
192.168.1.0    0.0.0.0         255.255.255.0   U     100    0        0 ens192
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
192.242.27.128  0.0.0.0         255.255.255.192 U     0      0        0 *
[root@k8s-node236-230 ~]# ip a |grep cali
````

But except 192.168.1.230, other nodes are working normally.

**Comment by dotbalo** [comment]
This problem has been solved: #4273
